.github/workflows/ci.yml

```diff
@@ -34,12 +34,42 @@ jobs:
           otp-version: ${{matrix.otp}}
           elixir-version: ${{matrix.elixir}}
 
+      - name: Restore dependencies cache
+        uses: actions/cache@v2
+        with:
+          path: deps
+          key: ${{ matrix.otp }}-${{ runner.elixir }}-mix-${{ hashFiles('**/mix.lock') }}
+          restore-keys: ${{ matrix.otp }}-${{ runner.elixir }}-mix-
+
       - name: Install Dependencies
         run: mix deps.get --only test
 
+      - name: Check formatting
+        run: mix format --check-formatted
+
       - name: Compile library
         run: mix compile --warnings-as-errors
 
       - run: mix credo --strict
 
+      - name: Retrieve PLT cache
+        uses: actions/cache@v2
+        id: plt-cache
+        with:
+          path: priv/plts
+          key: ${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-plts-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}
+
+      - name: Create PLTs
+        if: steps.plt-cache.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p priv/plts
+          mix dialyzer --plt
+
+      - name: Run dialyzer
+        run: mix dialyzer --no-check
+
+      - run: mix test --warnings-as-errors
+        if: matrix.elixir == '1.12.1'
```

Comment:
Why it's only this version?

Reply:
`mix test` with `--warnings-as-errors` only works with Elixir 1.12+
I tried with `matrix.elixir` >= 1.12 but it does not work

Reply:
Ah, that's the case. Sure - anyway thanks! :)